cmake/create_test.cmake

```diff
@@ -158,60 +155,72 @@ ginkgo_build_test_name(${test_name} test_target_name)
-        # set up actual test
-        set(test_target_name ${test_target_name}_${exec})
-        add_executable(${test_target_name} ${test_name}.cpp)
-        target_compile_features(${test_target_name} PUBLIC cxx_std_14)
-        target_compile_options(${test_target_name} PRIVATE ${GINKGO_COMPILER_FLAGS})
-        target_compile_definitions(${test_target_name} PRIVATE EXEC_TYPE=${exec_type} EXEC_NAMESPACE=${exec} GKO_COMPILING_${exec_upper})
-        target_link_libraries(${test_target_name} PRIVATE ${common_test_ADDITIONAL_LIBRARIES})
-        # use float for DPC++ if necessary
-        if((exec STREQUAL "dpcpp") AND GINKGO_DPCPP_SINGLE_MODE)
-            target_compile_definitions(${test_target_name} PRIVATE GINKGO_COMMON_SINGLE_MODE=1)
-            target_compile_definitions(${test_target_name} PRIVATE GINKGO_DPCPP_SINGLE_MODE=1)
-        endif()
-        ginkgo_set_test_target_properties(${test_name}_${exec} ${test_target_name})
-    endforeach()
+        ginkgo_create_common_test_internal(${test_name} DpcppExecutor dpcpp ${ARGN})
+    endif()
 endfunction(ginkgo_create_common_test)
 
-function(ginkgo_create_common_and_reference_test test_name)
-    ginkgo_create_common_test(${test_name})
+function(ginkgo_create_common_test_internal test_name exec_type exec)
+    cmake_parse_arguments(PARSE_ARGV 1 common_test "" "MPI_SIZE" "DISABLE_EXECUTORS;ADDITIONAL_LIBRARIES;ADDITIONAL_INCLUDES")
+    if(exec IN_LIST common_test_DISABLE_EXECUTORS)
+        return()
+    endif()
     ginkgo_build_test_name(${test_name} test_target_name)
-    set(test_target_name ${test_target_name}_reference)
+    string(TOUPPER ${exec} exec_upper)
+    # set up actual test
+    set(test_target_name ${test_target_name}_${exec})
     add_executable(${test_target_name} ${test_name}.cpp)
-    target_compile_features(${test_target_name} PUBLIC cxx_std_14)
-    target_compile_options(${test_target_name} PRIVATE ${GINKGO_COMPILER_FLAGS})
-    target_compile_definitions(${test_target_name} PRIVATE EXEC_TYPE=ReferenceExecutor EXEC_NAMESPACE=reference)
-    target_link_libraries(${test_target_name} PRIVATE ${ARGN})
-    ginkgo_set_test_target_properties(${test_name}_reference ${test_target_name})
-endfunction()
+    target_compile_definitions(${test_target_name} PRIVATE EXEC_TYPE=${exec_type} EXEC_NAMESPACE=${exec} GKO_COMPILING_${exec_upper})
+    target_link_libraries(${test_target_name} PRIVATE ${common_test_ADDITIONAL_LIBRARIES})
+    # use float for DPC++ if necessary
+    if((exec STREQUAL "dpcpp") AND GINKGO_DPCPP_SINGLE_MODE)
+        target_compile_definitions(${test_target_name} PRIVATE GINKGO_COMMON_SINGLE_MODE=1)
+        target_compile_definitions(${test_target_name} PRIVATE GINKGO_DPCPP_SINGLE_MODE=1)
+    endif()
+    ginkgo_set_test_target_properties(${test_target_name} ${ARGN})
+    ginkgo_add_test(${test_name}_${exec} ${test_target_name} ${ARGN})
+endfunction(ginkgo_create_common_test_internal)
+
+function(ginkgo_create_common_device_test test_name)
```

Comment:
Maybe a couple documentations on the aim of some of these functions to clarify what they do at a glance? Without detailing the parameters